### ISSUE TITLE
support for non-client-side-applications

### ISSUE THREAD

**Issue by boneskull** [nickjanssen/angus#8]
This seems like a good idea.  

Why is it limited to client-side code?

Can I configure Angus in a way that would allow me to use it for NPM libraries I write?

Is it even usable in this context?  Would someone developing my module have to install Angus to build it?  Or is this really just for production applications?  I don't understand if Angus is required to build Angus applications, or if each application has its own `Gruntfile.js` configured, and Angus just keeps them updated?

I have the _exact_ same problem with my server-side code; many Gruntfiles, many configs, old tasks fall out of favor.  Tried writing a Yeoman generator, but that's a pain and modules fall out-of-date too quickly.

**Comment by nickjanssen** [comment]
What build tasks do you require for your server? I'm curious, in my experience build tasks are not often needed for servers as you just need to run them. I do happen to be working on a web app myself however which will need an API to go along with it. It might be an idea to add a `server/` folder in the root of your app and let Angus run a specified number of tasks in that folder.

Regarding your question, what do you mean with 'module'? An Angus app? You don't need to configure a `Gruntfile.js` yourself, you just add a  `angus.config.js` and run `angus dev` from the app directory. So yes, you need to have Angus installed globally when you want to build an app. It might also be possible to install angus as an npm module, I haven't tried it out yet but it should work out of the box.

**Comment by boneskull** [comment]
Not a server.  NPM libraries.  Just modules.  Think something like `q` or even `grunt` itself, or a Grunt task.

If Angus could be used in this context, it'd have to change its behavior so that Angus was optional.  

Perhaps what I'm looking for is a somewhat different solution that doesn't exist yet.

**Comment by nickjanssen** [comment]
Right, but what kind of build tasks would you need for your module? Could you be more specific and give an example? It's meant specifically for frontend apps, and it relies on a `src/`, `bower_components` and `angus.config.js`.

Do you want to have custom build tasks and are tired of creating them over? If so, I am going to make an option soon where you can set custom tasks for angus to execute.

**Comment by boneskull** [comment]
@nickjanssen Yes.  Things like Grunt tasks to run tests, check jshint, generate documentation, etc., etc.
